public/js/app/leaderboards.js

```diff
@@ -51,185 +39,130 @@ function leaderboardUpdate(data) {
   });
 
   //determines the current page, whether to add or substract the missing players in case we pressed next or previous then it will add or substract players
-  let playerIndex = (data.length - 1) - (pageNumber * 100) - (addNextPlayer * plusOrMinusPlayerList); //- addNextPlayer;
-  let next100Players = data.length - 100 - (pageNumber * 100) - (addNextPlayer * plusOrMinusPlayerList);
+  let playerIndex = (playerList.length - 1) - (pageNumber * 100); //- addNextPlayer;
+  let next100Players = playerList.length - 101 - (pageNumber * 100);
   for (playerIndex; playerIndex > next100Players; playerIndex--) {
-    
-    
-    
-    
-    
-    //console.log(`playerIndex is ${playerIndex}`);
-    //console.log(`data.length minus missing players is ${data.length - missingPlayers}`);
-    //console.log(` for loop data.length is ${data.length - 100 - (pageNumber * 100)}`)
-    
 
     if (playerIndex <= 0) {
-      //console.log('I broke');
-      playerIndex -= (playerIndex) * -1;
-    } else {
-      console.log(` We didn't break! `);
+      console.log('I broke');
```

Comment:
"I broke" is not a valid error message. Try specifying exactly what went wrong, how to fix it and what to look for in the callstack. think of future developers who will have to debug your thing.